src/views/StatsView.vue

```diff
@@ -115,19 +118,39 @@ export default {
       return this.seasons.map((season) => {
         return {
           title: season.name,
-          value: season
-        }
+          value: season,
+        };
       });
-    }
+    },
+  },
+  methods: {
+    checkAndSelectSeason(seasonId) {
+      if (this.seasons.some(({ id }) => id === seasonId)) {
+        const [selectedSeason] = this.seasons.filter(({ id }) => id === seasonId);
+        this.selectedSeason = selectedSeason;
+      } else {
+        const [selectedSeason] = this.seasons;
+        this.selectedSeason = selectedSeason;
+      }
+    },
   },
   created() {
     this.loadingData = true;
     io.socket.get('/stats', (res) => {
-      this.seasons = res;
-      const [selectedSeason] = this.seasons
-      this.selectedSeason = selectedSeason;
-      this.loadingData = false;
+      if (res.length > 0) {
+        this.seasons = res;
+        const seasonId = parseInt(this.$route.params.seasonId);
+        this.checkAndSelectSeason(seasonId);
+      }
     });
+    this.loadingData = false;
+  },
```

Comment:
```js
created() {
  this.loadingData = true;

  io.socket.get('/stats', (res) => {
    if (!res.length) {
      // fast fail
      return;
    }
    this.seasons = res;
    const seasonId = parseInt(this.$route.params.seasonId);
    this.checkAndSelectSeason(seasonId);
  });
  
  this.loadingData = false;
},
```
Fast fails are preferred imo.

Also a little weird to me to set loadingData to true initially here, would it be better to have the default in the component state set to true and then just set to false when done? You may also want to wrap in a try catch because if the socket call fails in an uncaught way your false declaration at the bottom may never get called properly.